src/make_trade.ts

```diff
@@ -239,5 +241,12 @@ async function waitForTrade(
   const timeout = new Promise((resolve: (value: boolean) => void) => {
     setTimeout(resolve, TRADE_TIMEOUT_SECONDS * 1000, false);
   });
-  return await Promise.race([traded, timeout]);
+  // Events are not very reliable, so in case we didn't receive it we query the API
```

Comment:
Is this an issue with the `ethers` provider where events are getting lost?

Reply:
Either that or with the subscription on the Infura node.

Reply:
Maybe there is a way to do with with `eth_logs` polling :thinking:.